### PR TITLE
Feature/SLOP-90/Change User Preferences to User Keywords

### PR DESCRIPTION
**Pull request by marzahlde09**
## Describe your changes

-Preferences table dropped
-User Preference changed to User Keyword
-Foreign key dependencies updated
-Migration file was modified so User Preference was not dropped, and instead altered data

Create keywords
Create a User Keyword linking a user to a keyword and assigning a score
These connections should now be visible on each of the tables

## Issue ticket number and link

[Slop-90](https://tripleten-apiary.atlassian.net/jira/software/projects/SLOP/boards/2?selectedIssue=SLOP-90)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] If there are changes to custom resolvers, I have added / updated automated tests
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by bduckdev — APPROVED** [review]
nice work. does anything on the front end need to be updated along with this?

**Comment by marzahlde09** [comment]
I'm expecting front end changes to come later - user keywords are going to get mixed in with the "recommendations" as well of course! I figure making sure this works independently is a good first step

**Review by skovakina — CHANGES_REQUESTED** [review]
I believe we should discuss this change further. According to Figma and the concept of preferences, almost every keyword should be associated with a specific group of preferences. So, instead of deleting preferences, we should focus on connecting them to the keywords. 
image
We can also see this in the data: keywords, handicap, and KWType (aka preferences) that were deleted in this PR
image

**Comment by marzahlde09** [comment]
I followed the AC, and this should allow users to have scores for different keywords. The user-keyword table is connected to keywords, which has the keyword-type, so that relationship still exists.

**Review by skovakina — APPROVED** [review]
alright. I think you've done a good job here

File: migrations/20240529004854_change_user_preference_to_user_keyword/migration.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+  Warnings:
+
+  - You are about to drop the `preference` table. If the table is not empty, all the data it contains will be lost.
+
+*/
+-- DropForeignKey
+ALTER TABLE `userpreference` DROP FOREIGN KEY `UserPreference_preference_fkey`;
```
Comment (tomershvueli) [inline migrations/20240529004854_change_user_preference_to_user_keyword/migration.sql]: Let's fix the capitalization on these tables.

**Review by tomershvueli — DISMISSED** [review]
See 1 comment, and also let's sync with the latest from `main`.

**Review by Thatalent — APPROVED** [review]
LGTM

**Pull request merged**